client/landing/gutenboarding/onboarding-block/signup-form/index.tsx

```diff
@@ -72,4 +64,13 @@ const SignupForm = () => {
 	);
 };
 
+function renderTos() {
+	return __experimentalCreateInterpolateElement(
+		NO__( 'By creating an account you agree to our <link_to_tos>Terms of Service</link_to_tos>.' ),
+		{
+			link_to_tos: <ExternalLink href="https://wordpress.com/tos/" />,
+		}
+	);
+}
```

Comment:
I see we had a function before, but is there a reason to wrap this in a function call?

Can we do:

```tsx
const tosLink = __experimentalCreateInterpolateElement( … );
// …
return <Modal>{ tosLink }</Modal>;
```

right in the component body instead?

Reply:
> is there a reason to wrap this in a function call?

Just personal preference probably. I could argue that it makes the component's function body shorter; but it doesn't hide any branching so it's not removing any complexity from the function.

One of the (small) cons I see in hooks is that component function bodies can get long, so I'm always looking for ways to split things out if I can come up with a good name for it.

Hooks also bring a new way to write abstractions, so you can use them to shorten functions too, so it goes both ways :)